From c6cfe7c55f54504aa50bc6daca31aefde0d24264 Mon Sep 17 00:00:00 2001
From: Alex Crawford <alex.crawford@coreos.com>
Date: Tue, 27 Oct 2020 19:43:48 +0100
Subject: [PATCH 1/3] portage/repository/config.py: add disabled attribute to
 repos.conf

This flag allows a repos.conf file to disable a previously-defined repository.

https://bugs.gentoo.org/507284
---
 lib/portage/repository/config.py | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/lib/portage/repository/config.py b/lib/portage/repository/config.py
index 0b591d9..7b7f7a1 100644
--- a/lib/portage/repository/config.py
+++ b/lib/portage/repository/config.py
@@ -113,6 +113,7 @@ class RepoConfig:
         "clone_depth",
         "create_manifest",
         "disable_manifest",
+        "disabled",
         "eapi",
         "eclass_db",
         "eclass_locations",
@@ -318,6 +319,11 @@ class RepoConfig:
             location = None
         self.location = location
 
+        disabled = repo_opts.get('disabled')
+        if disabled is not None:
+            disabled = disabled.strip().lower() == 'true'
+        self.disabled = disabled or False
+
         missing = True
         self.name = name
         if self.location is not None:
@@ -845,7 +851,10 @@ class RepoConfigLoader:
         # Do this before expanding aliases, so that location_map and
         # treemap consistently map unaliased names whenever available.
         for repo_name, repo in list(prepos.items()):
-            if repo.location is None:
+            if repo.disabled:
+                del prepos[repo_name]
+                continue
+            elif repo.location is None:
                 if repo_name != "DEFAULT":
                     # Skip this warning for repoman (bug #474578).
                     if settings.local_config and paths:
-- 
2.32.0

